diff --git a/PlugIns/Assimp/include/OgreAssimpLoader.h b/PlugIns/Assimp/include/OgreAssimpLoader.h
index b631ee570..f7d6a25ec 100644
--- a/PlugIns/Assimp/include/OgreAssimpLoader.h
+++ b/PlugIns/Assimp/include/OgreAssimpLoader.h
@@ -92,6 +92,8 @@ public:
     AssimpLoader();
     virtual ~AssimpLoader();
 
+    static void SetAssimpImporterPostprocessFlags(int flags);
+
     bool load(const String& source, Mesh* mesh, SkeletonPtr& skeletonPtr,
               const Options& options = Options());
 
diff --git a/PlugIns/Assimp/src/AssimpLoader.cpp b/PlugIns/Assimp/src/AssimpLoader.cpp
index 0003a9b30..61ec8a6d7 100644
--- a/PlugIns/Assimp/src/AssimpLoader.cpp
+++ b/PlugIns/Assimp/src/AssimpLoader.cpp
@@ -330,6 +330,12 @@ String ReplaceSpaces(const String& s)
 } // namespace
 
 int AssimpLoader::msBoneCount = 0;
+int msImporterFlags = 0;
+
+void AssimpLoader::SetAssimpImporterPostprocessFlags(int flags)
+{
+    msImporterFlags = flags;
+}
 
 AssimpLoader::AssimpLoader()
 {
@@ -369,7 +375,7 @@ bool AssimpLoader::_load(const char* name, Assimp::Importer& importer, Mesh* mes
         flags &= ~aiProcess_GenNormals; // prefer smooth normals
 
     importer.SetPropertyFloat("PP_GSN_MAX_SMOOTHING_ANGLE", options.maxEdgeAngle);
-    const aiScene* scene = importer.ReadFile(name, flags);
+    const aiScene* scene = importer.ReadFile(name, flags | msImporterFlags);
 
     // If the import failed, report it
     if (!scene)
